src/FffGcodeWriter.cpp

```diff
@@ -496,15 +478,20 @@ void FffGcodeWriter::processLayer(SliceDataStorage& storage, int layer_nr, unsig
     }
 
     coord_t max_inner_wall_width = 0;
-    for (SettingsBaseVirtual& mesh_settings : storage.meshes)
+    for (const SettingsBaseVirtual& mesh_settings : storage.meshes)
     {
         max_inner_wall_width = std::max(max_inner_wall_width, mesh_settings.getSettingInMicrons((mesh_settings.getSettingAsCount("wall_line_count") > 1) ? "wall_line_width_x" : "wall_line_width_0")); 
     }
     int64_t comb_offset_from_outlines = max_inner_wall_width * 2;
 
+    assert(extruder_order_per_layer_negative_layers.size() + layer_nr >= 0 && "Layer numbers shouldn't get more negative than there are raft/filler layers");
+    const std::vector<unsigned int>& extruder_order =
+        (layer_nr < 0)?
+        extruder_order_per_layer_negative_layers[extruder_order_per_layer_negative_layers.size() + layer_nr]
+        :
+        extruder_order_per_layer[layer_nr];
```

Comment:
I found this better, because now we use the same index for the `extruder_order_per_layer` as the actual layer_nr. Otherwise it would get confusing and we could easily make mistakes.